process/block/baseProcess_test.go

```diff
@@ -1843,4 +1843,423 @@ func TestBaseProcessor_gasAndFeesDelta(t *testing.T) {
+
+		body := bp.GetFinalMiniBlocks(&block.MetaBlock{}, &block.Body{})
+		assert.Equal(t, &block.Body{}, body)
+	})
+
+	t.Run("scheduledMiniBlocks flag is set, empty body", func(t *testing.T) {
+		t.Parallel()
+
+		arguments := CreateMockArguments(createComponentHolderMocks())
+		arguments.ScheduledMiniBlocksEnableEpoch = 3
+		bp, _ := blproc.NewShardProcessor(arguments)
+		bp.EpochConfirmed(4, 0)
+
+		body := bp.GetFinalMiniBlocks(&block.MetaBlock{}, &block.Body{})
+		assert.Equal(t, &block.Body{}, body)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		arguments := CreateMockArguments(createComponentHolderMocks())
+		arguments.ScheduledMiniBlocksEnableEpoch = 3
+		bp, _ := blproc.NewShardProcessor(arguments)
+		bp.EpochConfirmed(4, 0)
+
+		mb1 := &block.MiniBlock{
+			TxHashes: [][]byte{[]byte("txHash1")},
+		}
+		mb2 := &block.MiniBlock{
+			TxHashes: [][]byte{[]byte("txHash2")},
+		}
+		body := &block.Body{
+			MiniBlocks: []*block.MiniBlock{
+				mb1,
+				mb2,
+			},
+		}
+
+		mbh1 := block.MiniBlockHeader{}
+		mbhReserved1 := block.MiniBlockHeaderReserved{State: block.Proposed}
+		mbh1.Reserved, _ = mbhReserved1.Marshal()
+
+		mbh2 := block.MiniBlockHeader{}
+		mbhReserved2 := block.MiniBlockHeaderReserved{State: block.Final}
+		mbh2.Reserved, _ = mbhReserved2.Marshal()
+
+		metaBlock := &block.MetaBlock{
+			MiniBlockHeaders: []block.MiniBlockHeader{
+				mbh1,
+				mbh2,
+			},
+		}
+
+		expectedBody := &block.Body{MiniBlocks: block.MiniBlockSlice{mb2}}
+
+		retBody := bp.GetFinalMiniBlocks(metaBlock, body)
+		assert.Equal(t, expectedBody, retBody)
+	})
+}
+
+func TestBaseProcessor_getScheduledMiniBlocksFromMe(t *testing.T) {
+	t.Parallel()
+
+	t.Run("wrong body type", func(t *testing.T) {
+		t.Parallel()
+
+		retBody, err := blproc.GetScheduledMiniBlocksFromMe(&block.Header{}, &wrongBody{})
+		assert.Equal(t, process.ErrWrongTypeAssertion, err)
+		assert.Nil(t, retBody)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		mb1 := &block.MiniBlock{
+			TxHashes: [][]byte{[]byte("txHash1")},
+		}
+		mb2 := &block.MiniBlock{
+			TxHashes: [][]byte{[]byte("txHash2")},
+		}
+		body := &block.Body{
+			MiniBlocks: []*block.MiniBlock{
+				mb1,
+				mb2,
+			},
+		}
+
+		mbh1 := block.MiniBlockHeader{
+			SenderShardID: 1,
+		}
+		mbhReserved1 := block.MiniBlockHeaderReserved{ExecutionType: block.Normal}
+		mbh1.Reserved, _ = mbhReserved1.Marshal()
+
+		mbh2 := block.MiniBlockHeader{
+			SenderShardID: 1,
+		}
+		mbhReserved2 := block.MiniBlockHeaderReserved{ExecutionType: block.Scheduled}
+		mbh2.Reserved, _ = mbhReserved2.Marshal()
+
+		header := &block.Header{
+			ShardID: 1,
+			MiniBlockHeaders: []block.MiniBlockHeader{
+				mbh1,
+				mbh2,
+			},
+		}
+
+		retBody, err := blproc.GetScheduledMiniBlocksFromMe(header, body)
+		assert.Nil(t, err)
+		assert.Equal(t, block.MiniBlockSlice{mb2}, retBody)
+	})
+}
+
+func TestBaseProcessor_checkScheduledMiniBlockValidity(t *testing.T) {
```

Comment:
An additional check should be made after the latest push in the main branch:
```
	if len(scheduledMiniBlocks) > len(headerHandler.GetMiniBlockHeadersHashes()) {
		log.Debug("baseProcessor.checkScheduledMiniBlocksValidity", "num mbs scheduled", len(scheduledMiniBlocks), "num mbs received", len(headerHandler.GetMiniBlockHeadersHashes()))
		return process.ErrScheduledMiniBlocksMismatch
	}
```

Reply:
added additional check